apps/api/src/applicant/ienapplicant.service.ts

```diff
@@ -535,4 +538,32 @@ export class IENApplicantService {
     };
     return this.ienapplicantJobRepository.findAndCount(query);
   }
+
+  async assignApplicant(id: string, employee: EmployeeRO) {
+    const applicant = await this.ienapplicantRepository.findOne(id);
+
+    const found = applicant?.recruiters?.some(e => e.id === employee.id);
+    if (found) return;
+
+    const haPcn = await this.haPcnRepository.findOne({ title: employee.organization });
+    if (!haPcn) {
+      throw new BadRequestException(`User doesn't belong to a health authority`);
+    }
+
+    let recruiterAssignment = await this.recruiterRepository.findOne({
+      applicant_id: id,
+      ha_id: haPcn.id,
+    });
+
+    if (!recruiterAssignment) {
+      recruiterAssignment = await this.recruiterRepository.create({
+        applicant_id: id,
+        ha_id: haPcn.id,
+        employee_id: employee.id,
+      });
+    } else {
+      recruiterAssignment.employee_id = employee.id;
+    }
+    return await this.recruiterRepository.save(recruiterAssignment);
```

Comment:
should we wrap this in a transaction incase it fails?

Reply:
I don't think we need a transaction for a single write query.